jscomp/core/lam_primitive.ml

```diff
@@ -267,12 +272,19 @@ let eq_primitive_approx ( lhs : t) (rhs : t) =
    | Pccall {prim_name = n1; prim_native_name = nn1; _} 
     -> n0 = n1 && nn0 = nn1 
    | _ -> false )    
-  | Pfield (n0, _dbg_info0) ->  (match rhs with Pfield (n1, _dbg_info1) ->  n0 = n1  | _ -> false )    
-  | Psetfield(i0, _dbg_info0) -> (match rhs with Psetfield(i1, _dbg_info1) ->  i0 = i1  | _ -> false)
+  | Pfield (n0, info0) ->  
+    (match rhs with Pfield (n1, info1) ->  n0 = n1 && eq_field_dbg_info info0 info1 | _ -> false )    
+  | Pfloatfield (i0, info0) -> 
+    (match rhs with Pfloatfield (i1,info1) -> i0 = i1 && eq_field_dbg_info info0 info1  | _ -> false)
+  | Psetfield(i0, info0) -> 
+    (match rhs with Psetfield(i1, info1) ->  i0 = i1 && eq_set_field_dbg_info info0 info1 | _ -> false)  
+  | Psetfloatfield (i0, info0) ->  
+    (match rhs with Psetfloatfield(i1,info1) -> i0 = i1 && eq_set_field_dbg_info info0 info1 | _ -> false)
+  | Pmakeblock (i, info0, flag0) -> 
+    (match rhs with Pmakeblock(i1,info1,flag1) ->  
+      i = i1 && flag0 = flag1 && info0 = info1 | _ -> false)  
```

Comment:
`info0 = info1` should this be a function like above

Reply:
done